### PR TITLE
Add functions to get used and free heap size

### PR DESCRIPTION
**Pull request by arjanmels**
- Added functionality to retrieve used and free heap size
- Refactored shared layout alignment code from alloc and dealloc

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "linked_list_allocator"
-version = "0.8.3"
+version = "0.8.4"
```
Comment (phil-opp) [inline Cargo.toml:3]: I normally do the version bumping as part of a separate publish commit, so could you please remove the version bump from this PR?

File: Changelog.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# 0.8.4
+
+- Add function to get used and free heap size ([#32](https://github.com/phil-opp/linked-list-allocator/pull/32))
+
```
Comment (phil-opp) [inline Changelog.md:4]: Thanks for directly adding a changelog entry, this is useful! Could you remove the `# 0.8.4` line since I normally do the version increment in a separate commit?

File: src/lib.rs [hunk 3]
```diff
@@ -67,25 +71,35 @@ impl Heap {
...
         }
     }
 
-    /// Allocates a chunk of the given size with the given alignment. Returns a pointer to the
-    /// beginning of that chunk if it was successful. Else it returns `None`.
-    /// This function scans the list of free memory blocks and uses the first block that is big
-    /// enough. The runtime is in O(n) where n is the number of free blocks, but it should be
-    /// reasonably fast for small allocations.
-    pub fn allocate_first_fit(&mut self, layout: Layout) -> Result<NonNull<u8>, ()> {
+    pub fn align_layout(&self, layout: Layout) -> Layout {
```
Comment (phil-opp) [inline src/lib.rs:80]: I think this could be a function without a `&self` argument, given that it does not need access to `self`. Also, could you please either add some basic documentation for this function or make it private (so that we don't have undocumented functions in our public API)?

File: Cargo.toml (hunk 1 above)
Comment (arjanmels) [reply Cargo.toml]: Sure. (Thought to make it easy, but it makes sense.)

File: Changelog.md (hunk 2 above)
Comment (arjanmels) [reply Changelog.md]: Done.

File: src/lib.rs (hunk 3 above)
Comment (arjanmels) [reply src/lib.rs]: Did both.

**Comment by phil-opp** [comment]
Published as version 0.8.4

**Comment by arjanmels** [comment]
Thanks for the quick publication.
